src/transformers/integrations/peft.py

```diff
@@ -517,3 +518,49 @@ def _dispatch_accelerate_model(
             offload_dir=offload_folder,
             **dispatch_model_kwargs,
         )
+
+    def delete_adapters(self, adapter_names: Union[List[str], str]):
+        """
+        Delete an adapter's LoRA layers from the underlying model.
+    
+        Args:
+            adapter_names (`Union[List[str], str]`):
+                The names (single string or list of strings) of the adapter to delete.
+    
+        Example:
+    
+        ```py
+        from diffusers import AutoPipelineForText2Image
+        import torch
+    
+        pipeline = AutoPipelineForText2Image.from_pretrained(
+            "stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16
+        ).to("cuda")
+        pipeline.load_lora_weights(
+            "jbilcke-hf/sdxl-cinematic-1", weight_name="pytorch_lora_weights.safetensors", adapter_names="cinematic"
+        )
+        pipeline.delete_adapters("cinematic")
+        ```
+        """
+    
+        if isinstance(adapter_names, str):
+            adapter_names = [adapter_names]
+    
+        for adapter_name in adapter_names:
+            for module in self.modules():
+                if isinstance(module, BaseTunerLayer):
+                    if hasattr(module, "delete_adapter"):
+                        module.delete_adapter(adapter_name)
+                    else:
+                        raise ValueError(
+                            "The version of PEFT you are using is not compatible, please use a version that is greater than 0.6.1"
+                        )
+    
+            # For transformers integration - we need to pop the adapter from the config
+            if getattr(self, "_hf_peft_config_loaded", False) and hasattr(self, "peft_config"):
+                self.peft_config.pop(adapter_name, None)
+                # In case all adapters are deleted, we need to delete the config
+                # and make sure to set the flag to False
+                if len(self.peft_config) == 0:
+                    del self.peft_config
+                    self._hf_peft_config_loaded = None
```

Comment:
```suggestion
                    self._hf_peft_config_loaded = False
```
Should keep it consistent with the initial state.